library/ssl_tls13_generic.c

```diff
@@ -994,15 +983,15 @@ static int ssl_tls13_write_certificate_verify_body( mbedtls_ssl_context *ssl,
     p += 2;
 
     /* Hash verify buffer with indicated hash function */
-    md_info = mbedtls_md_info_from_type( md_alg );
-    if( md_info == NULL )
-        return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
+    psa_algorithm = mbedtls_psa_translate_md( md_alg );
 
-    ret = mbedtls_md( md_info, verify_buffer, verify_buffer_len, verify_hash );
-    if( ret != 0 )
+    if( psa_hash_compute( psa_algorithm,
+                          verify_buffer,
+                          verify_buffer_len,
+                          verify_hash,sizeof( verify_hash ),
+                          &verify_hash_len ) != PSA_SUCCESS )
         return( ret );
```

Comment:
This should get and return `psa_ssl_status_to_mbedtls( status )` here instead of `ret`.